From a1fd6ac7a07ee336dfaf46b2caede08a5ddee2cb Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:05 -0400
Subject: [char] random: export add_disk_randomness

Message-id: <1402670238-13916-18-git-send-email-snitzer@redhat.com>
Patchwork-id: 83773
O-Subject: [RHEL7.1 PATCH 017/150] random: export add_disk_randomness
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit bdcfa3e57c9d92b082d2378bc9a64a3a8750fa8d
Author: Christoph Hellwig <hch@infradead.org>
Date:   Fri Apr 25 00:36:37 2014 -0700

    random: export add_disk_randomness

    This will be needed for pending changes to the scsi midlayer that now
    calls lower level block APIs, as well as any blk-mq driver that wants to
    contribute to the random pool.

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Acked-by: "Theodore Ts'o" <tytso@mit.edu>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/random.c b/drivers/char/random.c
index 7ccb4a0..0672be7 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -797,6 +797,7 @@ void add_disk_randomness(struct gendisk *disk)
 
  add_timer_randomness(disk->random, 0x100 + disk_devt(disk));
 }
+EXPORT_SYMBOL_GPL(add_disk_randomness);
 #endif
 
 /*********************************************************************
-- 
1.7.1